lib/Backend/LowerMDShared.cpp

```diff
@@ -268,6 +268,39 @@ LowererMD::LowerAsmJsCallE(IR::Instr * callInstr)
 }
 
 IR::Instr *
+LowererMD::LowerWasmMemOp(IR::Instr * instr, IR::Opnd *addrOpnd)
+{
+    IR::LabelInstr * helperLabel = Lowerer::InsertLabel(true, instr);
+    IR::LabelInstr * loadLabel = Lowerer::InsertLabel(false, instr);
+    IR::LabelInstr * doneLabel = Lowerer::InsertLabel(false, instr);
+
+    // Find array buffer length
+    IR::RegOpnd * indexOpnd = addrOpnd->AsIndirOpnd()->GetIndexOpnd();
+    IR::RegOpnd * baseOpnd = addrOpnd->AsIndirOpnd()->GetBaseOpnd();
+    Assert(baseOpnd->m_sym->IsSingleDef());
+    IR::Instr *defInstr = baseOpnd->m_sym->m_instrDef;
+    IR::RegOpnd *arrayBuffer = defInstr->GetSrc1()->AsIndirOpnd()->GetBaseOpnd();
+    IR::Opnd *srcOpnd = IR::IndirOpnd::New(arrayBuffer, Js::ArrayBuffer::GetByteLengthOffset(), TyMachReg, m_func);
+    IR::RegOpnd *arrayLenOpnd = IR::RegOpnd::New(TyMachReg, m_func);
```

Comment:
I don't understand, src2 of which instruction ? I am reading it from the Js::ArrayBuffer::GetByteLengthOffset() right now.